packages/react-api/src/hooks/useViewportFeatures.js

```diff
@@ -90,8 +101,14 @@ export default function useViewportFeatures(
 
   useEffect(() => {
     if (sourceId && tiles.length && (!isSourceV3 || isSourceTileset)) {
+      clearDebounce();
```

Comment:
You apply the clearDebounce just for tiles, what would happen to geojson? Doesn't it make sense to apply same approach?

Reply:
No, because geojson is only loaded once and only have one way to be updated: viewport changes. For when viewport changes fastly, we have debounced the function for filtering by viewport.
When we have tiles, the viewport features can be updated when viewport changes and when new tiles are loaded. That's why we need clearDebounce only in tiles, to stop debounce if we know that a change in the feature is coming.

Reply:
That looks reasonable.